sdk/scheduler/src/main/java/com/mesosphere/sdk/state/GoalStateOverride.java

```diff
@@ -0,0 +1,144 @@
+package com.mesosphere.sdk.state;
+
+import java.util.Arrays;
+
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+
+import com.mesosphere.sdk.specification.GoalState;
+
+/**
+ * The definition of a goal state override for a task. These are custom states, other than {@link GoalState}s, which a
+ * task may enter. The main difference between this and {@link GoalState} is that {@link GoalState}s are configured by
+ * the service developer, whereas {@link GoalStateOverride}s are applied by the operator.
+ */
+public enum GoalStateOverride {
```

Comment:
What's the value of making this an `enum`?  Not saying it's a bad idea, just looks a whole lot like a class to me.

Reply:
Mainly just for the freebie identity handling. An object-based system would involve likely involve a registry or something. Doing that would be more flexible in that developers could specify their own overrides, but I don't think we're to that point yet (would imply a well-defined set of 'hooks' where an override would take effect).